### ISSUE TITLE
undefined index db-ssl-verify

### ISSUE THREAD

**Issue by DanielRuf** [magento/magento2#26762]
### Preconditions (*)

1. current 2.4-develop
2. ddev setup

### Steps to reproduce (*)

Follow the install instructions at https://ddev.readthedocs.io/en/latest/users/cli-usage/#magento-2-quickstart and then try to change the backend frontend name using the ssh container.
```
bin/magento setup:config:set --backend-frontname=admin
Overwrite the existing configuration for backend-frontname?[Y/n]y
Overwrite the existing configuration for db-ssl-verify?[Y/n]n

In ErrorHandler.php line 61:
                                                                                                                                        
  Notice: Undefined index: db-ssl-verify in /var/www/html/setup/src/Magento/Setup/Model/ConfigOptionsList/DriverOptions.php on line 49  
```

### Expected result (*)

1. Result with no errors

### Actual result (*)

1. 
![screen](https://user-images.githubusercontent.com/51679138/75664457-2cc40c80-5c7b-11ea-8a84-e3c0e51c2dfa.png)

**Comment by DanielRuf** [comment]
I'm working on it.

**Comment by gcampedelli** [comment]
Hi, I had the same problem while configuring Redis using Magento Dev Docs as a guide. 
Turns out thar by declaring these values in command line I could go further. 

 `bin/magento setup:config:set --page-cache=redis --page-cache-redis-server=127.0.0.1 --page-cache-redis-db=1` 
is the proposed command by Magento Dev Docs
I wrote it like this
 `bin/magento setup:config:set --page-cache=redis --page-cache-redis-server=127.0.0.1 --page-cache-redis-db=1  --db-ssl-verify=/path-to/mysql/my.pem` 
I don't remember exactly which where the missing index, but it was a total of 4. So you can declare each one in command line. For me it worked. like this
 `bin/magento setup:config:set --page-cache=redis --page-cache-redis-server=127.0.0.1 --page-cache-redis-db=1  --db-ssl-verify=/path-to/mysql/my.pem --db-(other index)=path-to --db-(another index)=path-to`

**Comment by DanielRuf** [comment]
I don't think that my ddev setup uses redis. So your issue might be a different one.

**Comment by gcampedelli** [comment]
Have you tried to declare those index in setup? I had to do that as well for each command line I used. The only difference is that I made the setup using wizard. You can also try to declare in env.php like this:

```
 'db' => [
        'table_prefix' => '',
        'connection' => [
            'default' => [
                'host' => '127.0.0.1',
                'dbname' => 'dbname',
                'username' => 'magento_user',
                'password' => 'your strong password',
                'active' => '1',
                'driver_options' => [
                    1007 => 'path-to private pem',
                    1008 => 'path to cert.pem',
                    1009 => 'path to ca.pem',
                    1014 => false
                ],
                'model' => 'mysql4',
                'engine' => 'innodb',
                'initStatements' => 'SET NAMES utf8;'
            ]
        ]
    ],
```
Otherwise, you can declare in your config:setup (any of them) . --db . flags for each of those indexes (1007, 1008, 1009 and 1014)

**Comment by DanielRuf** [comment]
> Otherwise, you can declare in your config:setup (any of them) . --db . flags for each of those indexes (1007, 1008, 1009 and 1014)

I mean a very specific feild which fails in a standard setup (ddev). I did nothing special. And not sure what these numbers are for. I have already opened a PR with a bugfix for such cases.

**Comment by DanielRuf** [comment]
> Otherwise, you can declare in your config:setup (any of them) . --db . flags for each of those indexes (1007, 1008, 1009 and 1014)

I mean a very specific field which fails in a standard setup (ddev). I did nothing special. And not sure what these numbers are for. I have already opened a PR with a bugfix for such cases.

**Comment by gcampedelli** [comment]
I don't this is a bug. 1007, 1008, 1009 and 1014 each one correspond to --db-ssl-verify --db-cert-verify, the problem which your are facing while configuring setup. So, after configuring setup, db-ssl-verify turns out to be 1007 in driver options inside your env.php file.

**Comment by DanielRuf** [comment]
> 1014 => false

This seems to be already set in the generated file. Normal setup and so on worked but not the specific command which shoudl nothing have to do with the field (just change one setting).

The change in the PR prevents this error.

**Comment by DanielRuf** [comment]
Please follow the described steps. Not sure if you are part of the Magento or Adobe teams or a contributor as I see no additional information.

<img width="242" alt="Bildschirmfoto 2020-02-07 um 23 58 54" src="https://user-images.githubusercontent.com/827205/74072022-d231ea80-4a05-11ea-98c4-9aa23e050815.png">

**Comment by gcampedelli** [comment]
I'm not from Magento team, I'm just sharing with you how I solved a similar error calling config:setup. I did some research on that new feature and I didn't open an issue. I just said for you to try to declare those index while using config:setup. The same values that are declared in your env.php. I just think it is missing in Dev Docs files. So, you PR is relevant if you think so, but then it is up to Magento Administrators accept it or not according to their milestones and plans.

**Comment by DanielRuf** [comment]
Your issue is a different one as my config:setup worked (part of the magento 2 setup with ddev). Just the config:set command caused this error in my case ;-)

**Comment by flancer64** [comment]
I install Magento 2.3.4 using composer:
```
composer --repository-url=https://repo.magento.com/ create-project \
  magento/project-community-edition=^2.3 ./mage
```

then I run `./bin/magento setup:install` to connect to existing DB (I upgrade my Mage instance):
```
./bin/magento setup:install \
  --admin-email="admin@store.com" \
  --admin-firstname="Store" \
  --admin-lastname="Admin" \
  --admin-password="..." \
  --admin-use-security-key="0" \
  --admin-user="admin" \
  --backend-frontname="backdoor" \
  --base-url-secure="https://my.magento.com/" \
  --base-url="http://my.magento.com/" \
  --db-host="localhost" \
  --db-name="magento" \
  --db-password="..." \
  --db-user="magento" \
  --key="..." \
  --session-save="db" \
  --use-rewrites="1" \
  --use-secure-admin="0" \
  --use-secure="0"
```

after that I have absolutely fresh installation and newly created `./app/etc/env.php` with:
```
    'db' => [
        ...
        'connection' => [
            'default' => [
                ...
                'driver_options' => [
                    1014 => false
                ]
            ]
        ]
    ]
```

Then I have this warning:
```
Overwrite the existing configuration for db-ssl-verify?[Y/n]
```
every time I run
```
$ ./bin/magento setup:config:set
```
even w/o options.

**Comment by DanielRuf** [comment]
Both PRs will make it more stable and prevent such cases or errors that are not catched and cause issues in a later step.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Alfa 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-32043` were created

**Issue Available**: @engcom-Alfa, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by brusMX** [comment]
Thanks @gcampedelli !
I also got this bug while on my install not specifying `db-ssl-key` nor `db-ssl-cert` or me the command that fixed my install was to put those flags with an empty string:

```bash
magento setup:config:set   \
--db-ssl-key=""  \
--db-ssl-cert=""  \
--db-ssl-ca=/path/tho/ca.pem  \
--db-ssl-verify  \
--session-save=db
```

**Comment by magento-engcom-team** [comment]
Hi @DanielRuf. Thank you for your report. 
The issue has been fixed in magento/magento2#26763 by @DanielRuf in 2.4-develop branch
Related commit(s):
 - [191463693a45eed85f6b427ed8b82620d0b1047b](https://github.com/magento/magento2/commit/191463693a45eed85f6b427ed8b82620d0b1047b)

The fix will be available with the upcoming 2.4.0 release.

**Comment by magento-engcom-team** [comment]
Hi @DanielRuf. Thank you for your report. 
The issue has been fixed in magento/magento2#30077 by @flancer64 in 2.4-develop branch
Related commit(s):
 - [5f9897575102a63501d6909f3ac5a11a19969a3d](https://github.com/magento/magento2/commit/5f9897575102a63501d6909f3ac5a11a19969a3d)
 - [cf1a716e7a40acc30e763b0b68ca2e6f9ba85cac](https://github.com/magento/magento2/commit/cf1a716e7a40acc30e763b0b68ca2e6f9ba85cac)
 - [5c23f4d24d6ff9c95f50ba3378498676ba548eb4](https://github.com/magento/magento2/commit/5c23f4d24d6ff9c95f50ba3378498676ba548eb4)
 - [23aa8067a9d5a5d99a7851d63bf5eb486508f168](https://github.com/magento/magento2/commit/23aa8067a9d5a5d99a7851d63bf5eb486508f168)

The fix will be available with the upcoming 2.4.2 release.
